src/Parser/RestoDruid/Modules/Features/StatWeights.js

```diff
@@ -0,0 +1,316 @@
+ * simply 1 / totalLeechRating. If the player does not already have Leech, things are much more difficult. I have not yet
+ * settled on an approach, and so this calculation is not yet implemented.
+ *
+ * Special Cases -
+ *
+ * A few special cases are considered. We're ignoring all Tranq healing, even ticks that don't overheal, under the assumption
+ * that the last ticks always overheal so what's the difference? I know this is a violation of the "would've" rule I detailed
+ * above. This decision may be revisited.
+ *
+ * Leech and Velens heals are disregarded for the purpose of the main stat's weights, as they scale with whatever procced them.
+ * It's easiest to just not count them.
+ *
+ * If a heal is not in the database of heals, it's assumed to scale only with Vers. This is generally a good assumption,
+ * and this is how most trinket heals behave.
+ */
+class StatWeights extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    critEffectBonus: CritEffectBonus,
+  };
+
+  concordanceAmount = 0; // TODO remove when stat tracker added
+
+  totalAdjustedHealing = 0; // total healing after excluding 'multiplier' spells like Leech / Velens
+
+  // These are the total healing that would be gained if their respective stats ratings were increased by one.
+  totalOneInt = 0;
+  totalOneCrit = 0;
+  totalOneHasteHpm = 0;
+  totalOneMastery = 0;
+  totalOneVers = 0;
+  totalOneLeech = 0;
+
+  on_initialized() {
+    this.concordanceAmount = 3700 + (this.combatants.selected.traitsBySpellId[SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_TRAIT.id] || 0) * 300; // TODO remove when stat tracker added
+  }
+
+  on_byPlayer_heal(event) {
+    const healVal = new HealingValue(event.amount, event.absorbed, event.overheal);
+    this._handleHeal(event, healVal);
+  }
+
+  on_byPlayer_absorbed(event) {
+    const healVal = new HealingValue(event.amount, 0, 0);
+    this._handleHeal(event, healVal);
+  }
+
+  _handleHeal(event, healVal) {
+    const spellId = event.ability.guid;
+    const target = this.combatants.getEntity(event);
+    if(target === null) {
+      return;
+    }
+
+    const amount = healVal.effective;
+    const spellInfo = getSpellInfo(event.ability.guid);
+
+    // region LEECH
+    // We have to calculate leech weight differently depending on if we already have any leech rating.
+    // Leech is marked as a 'multplier' heal, so we have to check it before we do the early return below
+    const hasLeech = this.combatants.selected.leechRating > 0; // TODO replace when dynamic stats
+    if(hasLeech && spellId === SPELLS.LEECH.id && !healVal.overheal) {
+      this.totalOneLeech += amount / this.combatants.selected.leechRating; // TODO replace when dynamic stats
+    } else if(!hasLeech) {
+      // TODO this will be a pain to implement
+    }
+    // endregion
+
+    if(!spellInfo.multiplier) {
+      this.totalAdjustedHealing += healVal.effective;
+    }
+
+    if(spellInfo.multiplier || spellInfo.ignored || healVal.overheal) {
```

Comment:
I preferred the old way of doing this in 3 separate if-statements; made it clearer they're unrelated and don't depend on each other.